core/spec/lib/spree/event/adapters/default_spec.rb

```diff
@@ -99,6 +99,50 @@ def toggle
 
             expect(dummy.box).to eq('foo')
           end
+
+          it 'adds the fired event with given caller location to the firing result object' do
+            bus = described_class.new
+            dummy = Class.new do
```

Comment:
I noticed this file includes 7 repetitions of this class definition, I'm thinking that we can define it only once and reuse it since tests use instances so I don't think we're changing the class state... what do you think? They're about 11 lines each, so we're adding about 70 duplicate lines.

Reply:
I tend to be quite a bigot when it comes not to sharing context between test examples (they should be basically about defining independent and isolated scenarios). Still, you're right, and I went too far! (I do have exceptions, and class definitions are one of them). So I've updated the code, and all tests use a `counter` class, which is pretty flexible for many situations. Thanks for pointing it out! It looks much tidier now! :tada: